Classes/2/Tutorial 2 Instructions-sherlock.md

```diff
@@ -0,0 +1,639 @@
+mkdir -p \$STR/female_1p
+cd \$STR
+
+STAR \$CmmnPrms \$AdtlPrms --genomeDir \$REF --outFileNamePrefix \$STR/female_1p/female.1p. \\
+--readFilesIn \$RDIR/female-trimmed-pair1.fastq \$RDIR/female-trimmed-pair2.fastq
+
+## Create index for the mapped bam file; this will make it easier to browse the mapped bam file in IGV browser
+samtools index \$STR/female_1p/female.1p.Aligned.sortedByCoord.out.bam
+
+## compress skwer-processed reads to save space.
+gzip --best \$RDIR/*
+
+## compress files that are needed for downstream analysis
+find \$STR -type f \( -name "*.out" -o -name "*.tab" -o -name "*.sjdb" -o -name "*.results" \) | xargs gzip -9
+
+## Cleanup and Copy back all important files
+if [ ! -d \$OWD/reads ];
+then
+mkdir -p \$OWD/{reads,STAResults}
+fi
+
+## Copy back important files
+cp -a \$STR/female_1p \$OWD/STAResults/
+cp -a \$RDIR/* \$OWD/reads/
+
+## Remove scratch directory
+rm -rf \$SCR
+exit 0
+EOF
+```
+
+#### Start mapping
+```bash
+bash STARfemale_scr_sherlock
+```
+
+#### Let's now use the reads that were cleaned up using skewer for one of the samples and do trancriptome based mapping using kallisto
+Run this in a separate terminal with the "singlecell" environment turned on as soon as you see a reads folder appear in $GROUP_HOME/BIOC281/Classes/2/ that contains male-trimmed-pair1.fastq.gz and male-trimmed-pair2.fastq.gz
+```bash
+cd $GROUP_HOME/BIOC281/Classes/2/
+mkdir kallistoResults && cd kallistoResults
+
+ln -sv $GROUP_HOME/BIOC281/Classes/2/reads/male*fastq.gz ./
+ln -sv  $GROUP_HOME/BIOC281/Classes/2/RefSeq_Oct2020/GenomeIndex/kallistoIndex/hg38.refSeq.transcriptome.ERCC.idx
+
+## Notice how we can use zcat to pass compressed reads into kallisto
+kallisto quant -t 4 -i hg38.refSeq.transcriptome.ERCC.idx -o output -b 100 <(zcat male-trimmed-pair1.fastq.gz) <(zcat male-trimmed-pair2.fastq.gz)
+```
+
+### Decompress STAR results
+```bash
+gzip -d $GROUP_HOME/BIOC281/Classes/2/STAResults/male_1p/male.1p.ReadsPerGene.out.tab.gz
+```
+
+## We're now going to read in and look at how the different aligners compare, please open BIOC281/Classes/2/Tutorial 2.ipynb
+When finished, please save the outputs of Tutorial 2.ipynb (File > Export Notebook As... > Export Notebook to HTML) when you have completed it and upload it to Canvas.
+
+## Request resources for the next class
+After saving Tutorial 2.ipynb, return to the original terminal window where you launched jupyter. Quit jupyter by pressing "cntrl+c" (or "command+c") twice.
+
+### Leave the current Sherlock allocation
+After running the command below, your terminal should switch back to the login node on Sherlock
+```bash
+exit
+```
+
+### Run salloc
+Start "tmux" session on Sherlock and then then request resources
+```bash
+tmux
+## You should see the green tmux bar. Now run salloc requesting resources for the next class
+salloc --ntasks-per-node=1 --cpus-per-task=4 --mem=30G --time=0-3:00:00 --begin="13:30:00 10/23/20" --qos=interactive srun --pty bash -i -l
```

Comment:
Sherlock's salloc does not have an interactive qos.

This line should be
` salloc --ntasks-per-node=1 --cpus-per-task=4 --mem=30G --time=0-3:00:00 --begin="13:30:00 10/23/20" --qos=normal srun --pty bash -i -l`